src/fiber/channel.rs

```diff
@@ -1281,36 +1274,95 @@ where
         Ok(())
     }
 
-    pub fn register_tlc_remove(
+    pub async fn register_retryable_tlc_remove(
         &self,
         state: &mut ChannelActorState,
         tlc_id: TLCId,
         reason: RemoveTlcReason,
     ) {
-        let command = RemoveTlcCommand {
-            id: tlc_id.into(),
-            reason: reason.clone(),
-        };
-        if let Ok(_) = self.handle_remove_tlc_command(state, command) {
-            return;
-        } else {
-            error!("Failed to remove tlc: {:?}, retry it later", &tlc_id);
-            state.tlc_state.set_tlc_pending_remove(tlc_id, reason);
-        }
+        state.tlc_state.set_tlc_pending_remove(tlc_id, reason);
+        self.check_and_apply_retryable_remove_tlcs(state).await;
+    }
+
+    pub async fn register_retryable_relay_tlc_remove(
+        &self,
+        state: &mut ChannelActorState,
+        tlc_id: u64,
+        channel_id: Hash256,
+        reason: RemoveTlcReason,
+    ) {
+        state
+            .tlc_state
+            .insert_relay_tlc_remove(channel_id, tlc_id, reason);
+        self.check_and_apply_retryable_remove_tlcs(state).await;
     }
 
-    pub fn check_tlc_setdown(&self, state: &mut ChannelActorState) {
+    pub async fn check_and_apply_retryable_remove_tlcs(&self, state: &mut ChannelActorState) {
         let pending_removes = state.tlc_state.get_pending_remove();
-        for (tlc_id, reason) in pending_removes.iter() {
-            let id: u64 = (*tlc_id).into();
-            let command = RemoveTlcCommand {
-                id,
-                reason: reason.clone(),
-            };
-            if let Ok(_) = self.handle_remove_tlc_command(state, command) {
-                state.tlc_state.remove_pending_remove_tlc(&tlc_id);
-            } else {
-                error!("Failed to remove tlc: {:?}, retry it later", &tlc_id);
+        for retryable_remove in pending_removes.iter() {
```

Comment:
we may use into_iter and remove the clone of line 1307 since get_pending_remove returns a cloned vec already.